### DISCUSSION TITLE
How can I create a subscription system and listen to the results of invoices in .net wpf

### DISCUSSION THREAD

**Discussion by xDreamms** [btcpayserver/btcpayserver#6421] [Technical Support] [answered]
I hosted BTCPayServer and I got url and api key now I want to create a system that gives licenses for my WPF app immediately after the user pays on btc pay server. How can I do that? How can I listen invoices results?

**Comment by dennisreimann** [comment, marked as answer]
You can either poll the status of an [invoice](https://docs.btcpayserver.org/API/Greenfield/v1/#operation/Invoices_GetInvoice) or use the [InvoiceSettled webhook](https://docs.btcpayserver.org/API/Greenfield/v1/#operation/Webhook_InvoiceSettled). Currently there is no support for websockets for Greenfield API.
